pandas/_libs/tslibs/conversion.pyx

```diff
@@ -355,6 +366,24 @@ cdef _TSObject convert_datetime_to_tsobject(datetime ts, object tz,
         obj.value += nanos
         obj.dts.ps = nanos * 1000
 
+    if tz is not None:
+        if is_utc(tz) or is_tzlocal(tz):
+            # TODO: think on how we can infer fold for local Timezone
+            # and adjust value for fold
+            pass
+        else:
+            trans, deltas, typ = get_dst_info(tz)
+
+            if typ == 'pytz' or typ == 'dateutil':
```

Comment:
typ in ['....']

Reply:
Done. Also everywhere else where there was a similar check.